### PR TITLE
Jetpack AI Client: Introduce scaffolding for a Jetpack AI client JS package

### PR DESCRIPTION
**Pull request by oskosk**
Adds new JS Package

TODO
* [x] - Update README
* [x] - Fix tests
* [x] - Create GitHub repo
* [ ] - Create package repo.

## Proposed changes:

* Adds N

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

Instructiongs coming

### CODE REVIEW

File: projects/js-packages/ai-client/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  )
+};
+```
+
+### Requesting a Completion from the Jetpack AI API
+
+You can request a completion from the Jetpack AI API using the `requestCompletion` function. This function takes a prompt and optionally a post ID as parameters and returns an instance of `SuggestionsEventSource`.
+
+```
+import { requestCompletion } from 'jetpack-ai-client';
```
Comment (dhasilva) [inline projects/js-packages/ai-client/README.md]:
```suggestion
import { requestCompletion } from '@automattic/jetpack-ai-client';
```

File: projects/js-packages/ai-client/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  console.log( "Received so far", event.detail );
+} );
+```
+
+### Requesting Images from the Jetpack AI API
+
+You can fetch images from Jetpack AI using the `requestImages` function. This function takes a prompt and a post ID as parameters and returns a promise that resolves to an array of base64 encoded images.
+
+```
+import { requestImages } from 'jetpack-ai-client';
```
Comment (dhasilva) [inline projects/js-packages/ai-client/README.md]:
```suggestion
import { requestImages } from '@automattic/jetpack-ai-client';
```

File: projects/js-packages/ai-client/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+### Using the SuggestionsEventSource Class
+
+The `SuggestionsEventSource` class is a wrapper around `EventSource` that emits events for each chunk of data received, when the stream is closed, and when a full suggestion has been received.
+
+You shouldn't need to instantiate this class. You get one of these by calling `requestCompletion()`.
+
+```
+import { requestCompletion } from 'jetpack-ai-client';
```
Comment (dhasilva) [inline projects/js-packages/ai-client/README.md]:
```suggestion
import { requestCompletion } from '@automattic/jetpack-ai-client';
```

File: projects/js-packages/ai-client/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+### Requesting a Token from the Jetpack Site
+
+You can request a token from the Jetpack site using the `requestCompletionAuthToken` function. This function returns a promise that resolves to an object containing the token and the blogId.
+
+This function behaves properly whether it's called from a Jetpack environment or a WordPress.com one.
+
+```
+import { requestCompletionAuthToken } from 'jetpack-ai-client';
```
Comment (dhasilva) [inline projects/js-packages/ai-client/README.md]:
```suggestion
import { requestCompletionAuthToken } from '@automattic/jetpack-ai-client';
```

**Comment by dhasilva** [comment]
I asked for help setting up a mirror repo: p1686688537939379-slack-CBG1CP4EN
With that we can update the composer file with it.

**Review by jeherve — APPROVED** [review]

**Pull request merged**